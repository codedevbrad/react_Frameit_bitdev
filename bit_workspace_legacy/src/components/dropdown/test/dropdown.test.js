import React from 'react';
import ReactDOM from 'react-dom';
import Dropdown from '../dropdown/index';
import { render , cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Dropdown />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders Dropdown correctly' , ( ) => {
  const { getByTestId } = render(<Dropdown dropdownText='click me'> </Dropdown> )
  expect( getByTestId('dropdown')).toHaveTextContent('click me' );
}); 

it('matches snapshot' , ( ) => {
  const tree = renderer.create(
    <Dropdown dropdownText="click me new" openOnRender={true}>
        <li> hey  </li>
        <li> dude </li>
    </Dropdown> 
  ).toJSON();  
  expect( tree ).toMatchSnapshot();
});