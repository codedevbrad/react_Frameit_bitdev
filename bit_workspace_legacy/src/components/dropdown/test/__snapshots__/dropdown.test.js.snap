// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="dropdown"
  data-testid="dropdown"
>
  <div
    className="dropdown_titleArea"
    onClick={[Function]}
  >
    <h1>
       
      click me new
        
    </h1>
  </div>
  <div
    className="dropdown-content dropdown__undefined"
  >
    <ul
      id="navigation"
    >
      <li>
         hey  
      </li>
      <li>
         dude 
      </li>
    </ul>
  </div>
</div>
`;
